### PR TITLE
setSelectionAsFindPattern updates

### PR DESCRIPTION
**Pull request by adzenith**
These update the <kbd>cmd+e</kbd> behavior to use the word under the cursor if no word is selected (like <kbd>cmd+d</kbd> does) and to not clear the find pattern if nothing's selected and there's no word under the cursor.

### CODE REVIEW

**Comment by benogle** [comment]
Can you add specs for this?

File: lib/find-view.coffee [hunk 1]
```diff
@@ -329,8 +329,9 @@ class FindView extends View
       editor.scrollToCursorPosition(center: true)
 
   setSelectionAsFindPattern: =>
-    pattern = @findModel.getEditor().getSelectedText()
-    @updateModel {pattern}
+    pattern = @findModel.getEditor()?.getSelectedText() or @findModel.getEditor()?.getWordUnderCursor()
+    if pattern
+      @updateModel {pattern}
```
Comment (benogle) [inline lib/find-view.coffee]: This should be 

```
@updateModel({pattern}) if pattern
```

File: lib/project-find-view.coffee [hunk 2]
```diff
@@ -266,6 +266,6 @@ class ProjectFindView extends View
 
   setSelectionAsFindPattern: =>
     editor = atom.workspace.getActivePaneItem()
-    if editor?.getSelectedText?
-      pattern = editor.getSelectedText()
+    pattern = editor?.getSelectedText() or editor?.getWordUnderCursor()
+    if pattern
       @findEditor.setText(pattern)
```
Comment (benogle) [inline lib/project-find-view.coffee]: Same. Postfix on the if

**Comment by adzenith** [comment]
Whoops! Forgot about those inline ifs. Thanks.

I might look into adding some specs, but it's crunch time at work so it wouldn't be for a while.

**Comment by adzenith** [comment]
Closing. Superseded by #376.

**Pull request closed without merging**